addon/mixins/run.js

```diff
@@ -118,11 +120,82 @@ export default Mixin.create({
    ```
 
    @method cancelTask
-   @param { Number } cancelId the id returned from the runTask call
+   @param { Number } cancelId the id returned from the runTask or scheduleTask call
    @public
    */
   cancelTask(cancelId) {
-    cancelTimer(cancelId);
+    if (typeof cancelId.cancel === 'function') {
+      cancelId.cancel();
+    } else {
+      cancelTimer(cancelId);
+    }
+  },
+
+  /**
+   Adds the provided function to the named queue to be executed at the end of the RunLoop.
+   The timer is properly canceled if the object is destroyed before it is invoked.
+
+   Example:
+
+   ```js
+   import Component from 'ember-component';
+   import ContextBoundTasksMixin from 'web-client/mixins/context-bound-tasks';
+
+   export default Component.extend(ContextBoundTasksMixin, {
+     init() {
+       this._super(...arguments);
+       this.scheduleTask('afterRender', () => {
+         console.log('This runs at the end of the run loop (via the afterRender queue) if this component is still displayed');
+       })
+     }
+   });
+   ```
+
+   @method scheduleTask
+   @param { Function } callback the callback to run at the provided time
+   @param { ...* } args arguments to pass to the callback
+   @param { Number } [timeout=0] the time in the future to run the callback
+   @public
+   */
+  scheduleTask(queueName, callbackOrName, ...args) {
+    assert(`Called \`scheduleTask\` without a string as the first argument on ${this}.`, typeof queueName === 'string');
+    assert(`Called \`scheduleTask\` on destroyed object: ${this}.`, !this.isDestroyed);
+
+    let task;
+    let type = typeof callbackOrName;
+    if (type === 'function') {
+      task = () => {
+        if (!this.isDestroyed) {
```

Comment:
This probably needs to be modified to use `this.isDestroying || this.isDestroyed`.

Reply:
👍